qcelemental/covalent_radii.py

```diff
@@ -0,0 +1,148 @@
+        ]  # yapf: disable
+
+        # add alternate names to help QC programs
+        for alias in aliases:
+            ident, units, value, comment = alias
+            self.cr[ident.capitalize()] = datum.Datum(ident, units, value, comment)
+
+    def __str__(self):
+        return "CovalentRadii(context='{}')".format(self.name)
+
+    def get(self, atom, return_tuple=False, units=None):
+        """Access a covalent radius for species `atom`.
+
+        Parameters
+        ----------
+        atom : int or str
+            Identifier for element or nuclide, e.g., `H`, `D`, `H2`, `He`, `hE4`.
+            In general, one value recommended for each element; however, certain other exact labels may be available.
+            ALVAREZ2008: C_sp3, C_sp2, C_sp, Mn_lowspin, Mn_highspin, Fe_lowspin, Fe_highspin, Co_lowspin, Co_highspin
+        units : str, optional
+            Units of returned value. `None` returns native units (ALVAREZ2008: angstrom).
+            Only relevant for ``return_tuple=False`` since ``True`` returns underlying data structure with native units.
+        return_tuple : bool, optional
+            See below.
+
+        Returns
+        -------
+        float
+            When ``return_tuple=False``, value of covalent radius. If multiple defined for element, returns largest.
+        qcelemental.Datum
+            When ``return_tuple=True``, Datum namedtuple with units, description, uncertainty, and value of covalent radius as Decimal (preserving significant figures).
+            If multiple defined for element, returns largest.
+
+        Raises
+        ------
+        NotAnElementError
+            If `atom` cannot be resolved into an element or nuclide or label.
+        DataUnavailableError
+            If `atom` is a valid element or nuclide but not one for which a covalent radius is available.
+
+        """
+        if atom in self.cr.keys():
+            # catch extra labels like 'C_sp3'
+            identifier = atom
+        else:
+            identifier = periodictable.to_E(atom)
+
+        try:
+            qca = self.cr[identifier]
+        except KeyError as e:
+            raise DataUnavailableError('covalent radius', identifier) from e
+
+        if return_tuple:
+            return qca
+        else:
+            return qca.to_units(units)
+
+    def string_representation(self):
+        """Print name, value, and units of all covalent radii."""
+
+        return datum.print_variables(self.cr)
+
+    def write_c_header(self, filename='covrad.h'):
+        """Write C header file defining covalent radii array."""
+
+        text = [
+            '#ifndef _qcelemental_covrad_h_',
+            '#define _qcelemental_covrad_h_',
+            '',
+            '/* This file is autogenerated from the QCElemental python module */',
+            '',
+            'const double covalent_radii[] = {',
+        ]
+
+        for el in periodictable.E[1:50]:
```

Comment:
TODO: fix this range

Reply:
Did you still need to fix this?

Reply:
Yes. I only saw it as I was packing up, so left myself a note. I think `periodictable` needs an iterator fn.